python/cudf/cudf/core/index.py

```diff
@@ -1891,6 +1895,34 @@ def _concat(cls, objs):
         result.name = name
         return result
 
+    @cached_property
+    def inferred_type(self):
+        """
+        Return a string of the type inferred from the values.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> idx = cudf.Index([1, 2, 3])
+        >>> idx
+        Index([1, 2, 3], dtype='int64')
+        >>> idx.inferred_type
+        'integer'
+        """
+        infer_type = str(self.dtype)
+        if infer_type == "object":
+            if len(self) == 0:
+                return "empty"
+            else:
+                return "string"
+        elif self._is_integer():
+            return "integer"
+        elif self._is_floating():
+            return "floating"
+        elif self._is_boolean():
+            return "boolean"
+        return infer_type
```

Comment:
IIRC `inferred_type` in pandas shouldn't really return `str(self.dtype)`. I would prefer to `raise NotImplementedError` if this line isn't meaningfully hit.